crypto/cmp/cmp_client.c

```diff
@@ -582,10 +583,10 @@ static int cert_response(OSSL_CMP_CTX *ctx, int sleep, int rid,
         return 0;
     if (!save_statusInfo(ctx, crep->status))
         return 0;
-    if (rid == -1) {
+    if (rid == OSSL_CMP_CERTREQID_NONE) {
         /* for OSSL_CMP_PKIBODY_P10CR learn CertReqId from response */
         rid = ossl_cmp_asn1_get_int(crep->certReqId);
-        if (rid == -1) {
+        if (rid != OSSL_CMP_CERTREQID_NONE) {
```

Comment:
The `!=` looks awry.  Or does the comment need changing?
Either way I'm confused.

Reply:
Good catch - the comment  has become invalid; I've updated it.